src/coreclr/vm/ilstubcache.cpp

```diff
@@ -202,6 +202,7 @@ MethodDesc* ILStubCache::CreateNewMethodDesc(LoaderHeap* pCreationHeap, MethodTa
 #ifdef FEATURE_ARRAYSTUB_AS_IL
     if (SF_IsArrayOpStub(dwStubFlags))
     {
+        pMD->m_dwExtendedFlags |= DynamicMethodDesc::nomdStepThroughStub;
```

Comment:
Array stubs should be treated as leaf methods. I do not think it makes sense to mark them as step through stubs.